lib/active_model/serializer.rb

```diff
@@ -80,21 +80,29 @@ def self.serializers_cache
 
     # @api private
     # Find a serializer from a class and caches the lookup.
-    # Preferentially returns:
-    #   1. class name appended with "Serializer"
-    #   2. try again with superclass, if present
-    #   3. nil
     def self.get_serializer_for(klass)
       return nil unless config.serializer_lookup_enabled
       serializers_cache.fetch_or_store(klass) do
-        # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
-        serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
-
-        if serializer_class
-          serializer_class
-        elsif klass.superclass
-          get_serializer_for(klass.superclass)
-        end
+        get_serializer_class(klass)
+      end
+    end
+
+    # @api private
+    # Preferentially returns:
+    #   1. class name appended with "Serializer"
+    #   2. try again with superclass, if present
+    #   3. (optional) runs `on_serializer_not_found` policy if serializer is not found
+    #   4. nil
+    def self.get_serializer_class(klass)
+      # NOTE(beauby): When we drop 1.9.3 support we can lazify the map for perfs.
+      serializer_class = serializer_lookup_chain_for(klass).map(&:safe_constantize).find { |x| x && x < ActiveModel::Serializer }
+
+      if serializer_class
+        serializer_class
+      elsif klass.superclass && ![BasicObject, Object].include?(klass.superclass)
```

Comment:
Not sure what you mean - I've added a [test](https://github.com/rails-api/active_model_serializers/pull/1949/commits/d6a287bc0f396084254aac17b3951b321553a11f) to see if this doesn't mess up POROs - please take a look if this is what you're concerned about.

This is of course under the assumption that if an object is not serializable then it should not trigger the policy.